src/axom/sidre/tests/spio/spio_parallel.cpp

```diff
@@ -809,6 +810,145 @@ TEST(spio_parallel, parallel_increase_procs)
 
 }
 
+TEST(spio_parallel, parallel_decrease_procs)
+{
+  int my_rank;
+  MPI_Comm_rank(MPI_COMM_WORLD, &my_rank);
+  int num_ranks;
+  MPI_Comm_size(MPI_COMM_WORLD, &num_ranks);
+
+  EXPECT_TRUE(num_ranks > my_rank);
+
+#ifdef AXOM_USE_HDF5
+  /*
+   * This tests the ability to read data when the run that is doing the
+   * reading has fewer processors than the run that created the files
+   * being read.  In this test we dump data using all ranks on the world
+   * communicator, and then to read the data we split the world communicator
+   * into smaller communicators in order to read data using only ranks 0
+   * and 1 (only rank 0 if running on <= 2 processors).
+   *
+   * This functionality only works with the sidre_hdf5 protocol, so this
+   * section of code is inside the #ifdef guards.
+   */
+
+  DataStore* ds = new DataStore();
+  Group* root = ds->getRoot();
+
+  Group* flds = root->createGroup("fields");
+  Group* flds2 = root->createGroup("fields2");
+
+  Group* ga = flds->createGroup("a");
+  Group* gb = flds2->createGroup("b");
+  ga->createViewScalar<int>("i0", 101*my_rank);
+  gb->createView("i1")->allocate(DataType::c_int(10));
+  int* i1_vals = gb->getView("i1")->getData();
+
+  for(int i=0 ; i<10 ; i++)
+  {
+    i1_vals[i] = (i+10) * (404-my_rank-i);
+  }
+
+  int num_files = num_ranks;
+  axom::sidre::IOManager writer(MPI_COMM_WORLD);
+
+  const std::string file_name = "out_spio_parallel_decrease_procs";
+
+  writer.write(root, num_files, file_name, PROTOCOL);
+
+  int top_input_rank = 1;
+  if (num_ranks <= 2)
+  {
+    top_input_rank = 0;
+  }
+
+  // Split the communicator so that ranks up to and including
+  // top_input_rank have their own communicator for the input step.
+  MPI_Comm split_comm;
```

Comment:
I think you need to call `MPI_Comm_free( split_comm )` at the end to free that communicator and properly cleanup any MPI state.